app/Model/Model.php

```diff
@@ -92,11 +91,10 @@ private function bindValues(PDOStatement $stmt, array $attr): void
         foreach ($attr as $key) {
             $parts = explode("_", $key);
             $getter = "get" . ucfirst($parts[0]) . ucfirst($parts[1] ?? "");
-            if (!$this->{$getter}() instanceof DateTime) {
+            if (!$this->{$getter}() instanceof DateTime || !is_array($this->{$getter}())) {
```

Comment:
in order to support types properly you need to check on each type alone and do what is necessary to cast data